include/deal.II/grid/cell_id.h

```diff
@@ -48,34 +52,61 @@ template <int, int> class Triangulation;
  *
  * @note How this data is internally represented is not of importance (and not
  * exposed on purpose).
- *
- * @todo Does it make sense to implement a more efficient representation
- * (internally and/or as a string)? If yes, something like a 64bit int as in
- * p4est would be a good option.
  */
 class CellId
 {
 public:
   /**
-   * Construct a CellId object with a given @p coarse_cell_index and list of child indices.
+   * A type that is used to encode the CellId data in a compact and fast way
+   * (e.g. for MPI transfer to other processes). Note that it limits the
+   * number of children that can be transferred to 20 in 3D and 30 in 2D
+   * (using 2 times 32 bit for storage), a limitation that is identical to
+   * the one used by p4est.
+   */
+  typedef std_cxx11::array<unsigned int, 4> binary_type;
+
+  /**
+   * Construct a CellId object with a given @p coarse_cell_index and vector of
+   * child indices.
    */
   CellId(const unsigned int coarse_cell_id,
          const std::vector<unsigned char> &child_indices);
 
   /**
+   * Construct a CellId object with a given @p coarse_cell_index and array of
+   * child indices provided in @p child_indices. @p child_indices shall have
+   * at least @p n_child_indices valid entries.
+   */
+  CellId(const unsigned int coarse_cell_id,
+         const unsigned int n_child_indices,
+         const unsigned char *child_indices);
```

Comment:
can you say in the docs how `child_indices` is interpreted? I imagine that each byte corresponds to a number between zero and the number of children per cell in whatever the space dimension is this object corresponds to?